src/main/java/somethingrandom/usecase/search/SearchItemsOutputBoundary.java

```diff
@@ -0,0 +1,27 @@
+package somethingrandom.usecase.search;
+
+import somethingrandom.entity.Item;
+
+import java.util.Collection;
+
+/**
+ * The interface for presenters that wish to be used to display search results.
+ */
+public interface SearchItemsOutputBoundary {
+    /**
+     * Called when the search completes successfully, along with the list of
+     * corresponding entities.
+     * <p>
+     * As they are provided in a Collection, there is no guarantee of order.
+     * It's expected that the view be responsible for this.
+     */
+    void presentSearchResults(Collection<Item> items);
```

Comment:
I believe we are not allowed to have entities outside of the interactor

Reply:
I know, but I asked the TA during lab and he said that it was OK on the output boundary but not on the input boundary. I also don't know what would be better to replace it with, since I think we'd just have a bunch of not-quite-entities dotted around (we already have AddItemInputData). Any ideas? I'll check my copy of Clean Architecture quickly too.

Reply:
> ### Which Data Crosses the Boundaries
> Typically the data that crosses the boundaries consists of simple data structures. [...] The important thing is that isolated, simple data structures are passed across the boundaries. **We don't want to cheat and pass Entity objects** or database rows. We don't want the data structures to have any kind of dependency that violates the Dependency Rule.

(emphasis mine)

That seems to imply we should make some output-data object. It does seem like a *lot* of repetition, though, and it is the format more convenient for the use-cases. Per the Dependency Rule, it also shouldn't be a problem, since the dependency *is* pointing inwards, towards the entities. (I'd also argue our entities *are* simple, isolated data structures.)

I guess currently our UUID choice could save us. This could be a collection of `SearchItemsOutputData` objects:

```java
public class SearchItemsOutputData {
    private final UUID uuid;
    private final String title;

    public UUID getUUID();
    public String getTitle();
}
```
and then rely on the view-details use-case for the rest. This would mean that sorting by e.g. date would be impossible unless the use-case did it, though.

Might have to see if I can go to office hours tomorrow about it, since it seems like it's either breaking the rules or doing a bunch of extra SOLID-defying work.

Reply:
Added a new type, create a new comment there if you have questions.